src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -516,174 +512,102 @@ private void GenerateWebConfigFileForWwwRootOut(PublishRoot root, Runtime.Projec
             {
-                httpRuntime = new XElement("httpRuntime");
-                systemWeb.Add(httpRuntime);
+                root.Reports.WriteWarning("No commands defined. Defaulting to web.");
             }
-            // There is an httpRuntime element. The user may have already set this attribute...
-            else if (httpRuntime.Attribute("targetFramework") != null)
+            else
             {
-                // User already had a target framework, leave it alone
-                return;
+                root.Reports.WriteWarning("Multiple commands defined. Defaulting to web.");
             }
-            // Ok, now we have an httpRuntime element and we know we need to set thet targetFramework on it.
 
-            var bestDnxVersion = project.GetTargetFrameworks()
-                .Where(f => f.FrameworkName.Identifier.Equals(FrameworkNames.LongNames.Dnx))
-                .OrderByDescending(f => f.FrameworkName.Version)
-                .Select(f => f.FrameworkName.Version)
-                .FirstOrDefault();
-            if (bestDnxVersion != null)
-            {
-                httpRuntime.SetAttributeValue("targetFramework", bestDnxVersion.ToString());
-            }
-        }
+            command = command ?? "web";
 
-        private static XElement GetOrAddElement(XElement parent, string name)
-        {
-            var child = parent.Elements().Where(x => x.Name == name).FirstOrDefault();
-            if (child == null)
-            {
-                child = new XElement(name);
-                parent.Add(child);
-            }
-            return child;
-        }
+            root.Reports.Information.WriteLine($"Using command '{command}' as the entry point for web.config.");
 
-        private static string GetBootstrapperVersion(PublishRoot root)
-        {
-            // Use version of Microsoft.AspNet.Loader.IIS.Interop as version of bootstrapper
-            var package = root.Packages.SingleOrDefault(
-                x => string.Equals(x.Library.Name, "Microsoft.AspNet.Loader.IIS.Interop"));
-            return package == null ? string.Empty : package.Library.Version.ToString();
-        }
+            var targetDocument = XDocument.Parse($@"<configuration><system.webServer>
+  <handlers>
+    <add name=""httpplatformhandler"" path=""*"" verb=""*"" modules=""httpPlatformHandler"" resourceType=""Unspecified"" />
+  </handlers>
+  <httpPlatform processPath=""..\{command}.cmd""
+                arguments="""" 
+                stdoutLogEnabled=""true"" 
+                stdoutLogFile=""..\logs\stdout.log"">
+  </httpPlatform>
+</system.webServer></configuration>");
 
-        // Expected runtime name format: dnx-{FLAVOR}-{OS}-{ARCHITECTURE}.{VERSION}
-        // Sample input: dnx-coreclr-win-x86.1.0.0.0
-        // Sample output: coreclr
-        private static string GetRuntimeFlavor(PublishRuntime runtime)
-        {
-            if (runtime == null)
+            var attributesToOverwrite = new[]
```

Comment:
<del>`attributesToOverwrite`</del> `requiredAttributes` maybe? Overwrite makes me think they are the attributes that the user *can* overwrite.